### DISCUSSION TITLE
Process huge TXT files

### DISCUSSION THREAD

**Discussion by numerys** [coqui-ai/TTS#1133] [General Q&A]
How do I process huge TXT files with 5 MB or more?

I did

    tts --model_name tts_models/en/sam/tacotron-DDC --text <huge_text.txt
    tts --model_name tts_models/en/sam/tacotron-DDC --text `cat huge_text.txt`

without success.

**Comment by numerys** [comment]
A

    tts --model_name tts_models/en/sam/tacotron-DDC --text "${txt}"

works but results in

    bash: /home/user/tts/bin/tts: Argument list too long

which is correct. A standard kernel won't accept such large commands.

**Reply by erogol** [reply]
Probably it is better to code your own script using the Synthesizer class for such bespoke cases.

**Comment by christosangelopoulos** [comment]
This is a little script I wrote that does just this:
[https://gitlab.com/christosangel/sapo](https://gitlab.com/christosangel/sapo)
